### PR TITLE
Replace AppVeyor and Travis with GitHub Actions

### PR DESCRIPTION
**Pull request by Margen67**
### Why?
Because AppVeyor is slow.

### CODE REVIEW

**Comment by YaLTeR** [comment]
Could you move the clang-format check from travis to GHA and remove travis and appveyor altogether? Also why are debug configs commented out?

**Comment by Margen67** [comment]
Yes.
Debug configs fail to build from a clean build: #72

**Comment by YaLTeR** [comment]
You can add the format job in the CI.yml file. Just put the whole format: section after build:. Also you can delete travis and appveyor configs altogether (and remove them from ignored paths in the yml).

**Comment by Margen67** [comment]
@YaLTeR I did that and also uncommented Debug.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            SDK: episode1
+            spt-filename: spt-oe
+          - Configuration: Debug P2
+            SDK: portal2
+            spt-filename: spt-p2
+    steps:
+    - name: Checkout AlliedModders SDK
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
```
Comment (YaLTeR) [inline .github/workflows/CI.yml]: Why? I read that the default is 1, which should be sufficient?

Comment (Margen67) [reply .github/workflows/CI.yml]: Does that apply here or in general?

Comment (YaLTeR) [reply .github/workflows/CI.yml]: In general.

Comment (YaLTeR) [reply .github/workflows/CI.yml]: (as long as it works)

**Comment by YaLTeR** [comment]
And one last thing, could you check that the formatting job does indeed fail the build? Try committing some formatting error.

**Comment by Margen67** [comment]
> 
> 
> And one last thing, could you check that the formatting job does indeed fail the build? Try committing some formatting error.

Like what?

**Comment by YaLTeR** [comment]
Idk, like adding a bunch of spaces between arguments to some function.

**Comment by Margen67** [comment]
Yup, formatting works: https://github.com/Margen67/SourcePauseTool/runs/433761083

**Comment by YaLTeR** [comment]
Thanks for porting the CI!

**Comment by YaLTeR** [comment]
hold up, it failed https://github.com/YaLTeR/SourcePauseTool/runs/434150058?check_suite_focus=true#step:6:10

**Comment by Margen67** [comment]
It succeeds on my local repo so that makes no sense.
Besides reverting this PR, either this should be converted to use cmake or updated to VS2019.

**Pull request merged**